### ISSUE TITLE
Bottom D2F microswitch already pressed once screwed in to twinsor front

### ISSUE THREAD

**Issue by Brownbear1984** [IRTrail/G2E-Filametrix#6]
Firstly I want to say I appreciate all the work you've done here. I'm hoping the below will make sense.

I've printed out the G2E_Filametrix_Twinsor_Front.STL 2 or 3 times now at different flow settings incase I was over or under extruding. Every time I test the fitment of the D2F and bearing in the bottom I can never actuate the switch when passing filament through. If I take the screws out and just hold the microswitch in place I can hear the "click" from the switch when passing filament through. It also looks like the holes are slightly off when I can reliably actuate the switch. The top one works fine on all the versions I've printed but the bottom may need either the bottom screws to move out slightly or the bearing to move in slightly for reliable actuation of the switch. 

For reference these are the microswitches I have. https://a.aliexpress.com/_EHaO0pT

**Comment by IRTrail** [comment]
Most of the issues I hear about with switches not working is a stack up of tolerances.
The data sheet on the Omron is pretty specific with only 0.5mm "pretravel" (from initial contact until switch engagement) and 0.25mm "overtravel" (how much the plunger can move after switching). I have measured different actuation distances on a few different switches, even in the same brand. I don't have genuine Omron, so they might have better manufacturing tolerances than budget brands.
I have had people report back that different switches (even from the same package) work and others have indicated that different screws have helped.
I suspect a switch with a quarter millimeter off on actuation distance along with holes that are a quarter millimeter out of diameter and possibly out of position along with screws that have threads stamped on an old die likely cause this issue for some.
Also, the ball should be 5.5mm. Those usually have pretty good tolerance, but the difference between a 5.5mm and a 5.0mm can be missed pretty easily especially if you bought an assortment like I did. This has been the cause of confusion as well. Double check with a caliper. I have tried, but my Mk I Eyeball measurement tool just doesn't work in this situation.

My advice is to try a few different switches just to see if one mates up better. I recently added a link to the switches I bought from Amazon (in the readme) and have had decent luck with them, only a couple have been out of tolerance enough to be relegated to endstop switches.

There was a fairly extensive series of tests to find the optimal distance with several different variables. The unfortunate truth is that it's really difficult to hold a quarter millimeter tolerance with a 3D printer.
